### PR TITLE
Fail when not specifying number of decomposition components to plot

### PR DESCRIPTION
**Pull request by mwalls**
to stop it exploding when you forget to put a number

### CODE REVIEW

**Comment by francisco-dlp** [comment]
Wouldn't it have been easier to set the number in [plot_decomposition_factors](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/hyperspy/signal.py#L740) and plot_decomposition_loading](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/hyperspy/signal.py#L863)? e.g. 

```python

    def plot_decomposition_loadings(self,
                                    comp_ids=3,
                                    calibrate=True,
                                    same_window=None,
                                    comp_label='Decomposition loading',
                                    with_factors=False,
                                    cmap=plt.cm.gray,
                                    no_nans=False,
                                    per_row=3):
```

**Comment by mwalls** [comment]
Yes of course. As you know, my programming style is to change random lines of code till it does what I want.  But I have two questions:
1) Does your way mean that the default is changed to 3 but if someone (for what reason I don't know) actually enters "None" they will still get all of them?
2) Ok we'll use your method but I have to do it, right? (If I do the pull request I've got to make the changes?). Still a bit new to this...

**Comment by mwalls** [comment]
Also, if anyone has strong feelings about it being something other than 3 components, I'm not going to fight about it.

**Comment by francisco-dlp** [comment]
I am actually wondering if it is not better not to set any default and let the function fail if the number of components is not specified. to do that simply remove the ``None`` from the function keyword argument.

You are not obliged to do it, but, since it's your idea and you've already at it...

The easiest way to proceed is to revert the changes, commit the new version and push e.g.

```bash

git revert 1535cd6

### change the code ###

git commit -a -m "Set default number of components for plotting"
git push

```

**Comment by ericpre** [comment]
+1 on letting fail the function if the number of components is not specified. I don't think that there is a sensible default.
If there is one thing that I remember of my early days of HyperSpy is the frustration just after realising that I forgot to specify the number of components... Now, I am very well vaccinated with that one! :D

**Comment by mwalls** [comment]
Ive just tried to do it with the error message but can't tell if you're all seeing it. Sorry.

File: hyperspy/signal.py [hunk 1]
```diff
@@ -374,7 +374,7 @@ def _plot_factors_or_pchars(self, factors, comp_ids=None,
         else:
             return f
 
-    def _plot_loadings(self, loadings, comp_ids=None, calibrate=True,
```
Comment (ericpre) [inline hyperspy/signal.py:377]: This was not necessary to change the definition of this function, but it does not have any consequences. It just makes the code a bit less let consistence with the definition of `_plot_factors_or_pchars`.

**Comment by ericpre** [comment]
Yes, this is good and it could be merged: there is just a minor comment, but it is not really important.
The tests failing in travis are not related to this PR.

**Comment by mwalls** [comment]
ok by me. Can't remember if I'm allowed to close one. Will try now.

**Comment by mwalls** [comment]
Looks like I shouldn't have done that.

**Comment by ericpre** [comment]
Yes, please just wait that somebody merges it!

**Pull request merged**